engine/orchestration/src/main/java/com/cloud/vm/VirtualMachinePowerStateSyncImpl.java

```diff
@@ -162,6 +177,8 @@ private void processReport(long hostId, Map<Long, VirtualMachine.PowerState> tra
                         }
 
                         _messageBus.publish(null, VirtualMachineManager.Topics.VM_POWER_STATE, PublishScope.GLOBAL, instance.getId());
+                        ActionEventUtils.onActionEvent(User.UID_SYSTEM, Account.ACCOUNT_ID_SYSTEM,instance.getDomainId(),
+                                EventTypes.EVENT_VM_STOP, "Out of band VM power off", instance.getId(), ApiCommandResourceType.VirtualMachine.toString());
```

Comment:
I think this event should be spewed at `handlePowerOffReportWithNoPendingJobsOnVM` and countered by a `EVENT_VM_START` in `handlePowerOnReportWithNoPendingJobsOnVM` both in `VirtualMachineManagerImpl`. Or alternatively a proper place to handle the OOB start event in this class should be found. I think this is not a big deal for you @mlsorensen , but for people using event/usgae server based billing it would be. If an OOB migration happens their billing for this VM would/could stop or continue based on the order of events.

Reply:
Understood, I'll look into that.

Reply:
This may get is what you're looking for @DaanHoogland - in testing this didn't trigger any event at all during out of band migration on KVM. However it still responded to OOB stop, and even OOB start, publishing an event in both situations.

<img width="1247" alt="Screenshot 2023-08-24 at 10 32 46 AM" src="https://github.com/apache/cloudstack/assets/1047709/aeaa8d39-2161-4d4f-9864-f7c179e2309a">

Reply:
@mlsorensen , this is very hard to reproduce as it depends on the order of power sync reports coming in from hosts. I'll give this a swing after 4.18.1, and think about simulating race conditions. code looks good.